sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -442,7 +442,11 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
   override def uncacheTable(tableName: String): Unit = {
     val tableIdent = sparkSession.sessionState.sqlParser.parseTableIdentifier(tableName)
     val cascade = !sessionCatalog.isTemporaryTable(tableIdent)
-    sparkSession.sharedState.cacheManager.uncacheQuery(sparkSession.table(tableName), cascade)
+    val wasCached = Try(sparkSession.catalog.isCached(tableIdent.unquotedString)).getOrElse(false)
+    if (wasCached) {
+      sparkSession.sharedState.cacheManager.uncacheQuery(sparkSession.table(tableIdent), cascade)
+      sparkSession.sessionState.catalog.dropTempView(tableIdent.table)
```

Comment:
If the table is not a temporary view, no need to drop it.

Reply:
@viirya `dropTempView`  will only delete the temporary view. Are you suggesting that I increase a judgment on the tempory view here?